application/models/services/QuestionGroupService.php

```diff
@@ -18,22 +19,31 @@ class QuestionGroupService
 {
     private Permission $modelPermission;
     private Survey $modelSurvey;
-    private CWebUser $modelUser;
-    private CHttpRequest $modelRequest;
+    private Question $modelQuestion;
+    private QuestionGroup $modelQuestionGroup;
+    private QuestionGroupL10n $modelQuestionGroupL10n;
+    private CWebUser $yiiUser;
```

Comment:
Do we need the prefix `yii`?  Same with `yiiRequest` below. What else would it be except yii?

Reply:
As I was discussing with Tim yesterday, some of these dependancies should not being injected at all. Dependancies with the "yii" prefix fall in to the category and the yii prefix helps us easily identify them as we move forward in future refactoring.

The application model to be portable and not dependant on any specific MVC framework code or any code relating to a specific environment in which it could be run. 

To factor out all reference to Yii code would massively increase the effort and time required to complete this refactoring round. 

In short the yii prefix highlights that the dependancies is MVC framework specific and should be factored out in future.